saleor/graphql/account/mutations/permission_group.py

```diff
@@ -107,118 +214,102 @@ class Meta:
-        error_type_field = "account_errors"
-
-
-class PermissionGroupAssignUsers(ModelMutation):
-    group = graphene.Field(Group, description="Group to which users were assigned.")
-
-    class Arguments:
-        id = graphene.ID(
-            description="ID of the group to which users will be assigned.",
-            required=True,
-        )
-        users = graphene.List(
-            graphene.NonNull(graphene.ID),
-            description="List of users to assign to this group.",
-            required=True,
-        )
-
-    class Meta:
-        description = "Assign users to group."
-        model = auth_models.Group
-        permissions = (AccountPermissions.MANAGE_STAFF,)
-        error_type_class = AccountError
-        error_type_field = "account_errors"
+        error_type_class = PermissionGroupError
+        error_type_field = "permission_group_errors"
 
     @classmethod
-    def perform_mutation(cls, root, info, **data):
+    @transaction.atomic
+    def perform_mutation(cls, _root, info, **data):
         group = cls.get_instance(info, **data)
-        user_pks = cls.prepare_user_pks(info, group, **data)
-        cls.check_if_users_are_staff(user_pks)
-        group.user_set.add(*user_pks)
+        if not can_user_manage_group(info.context.user, group):
+            raise PermissionDenied()
+        data = data.get("input")
+        cleaned_input = cls.clean_input(info, group, data)
+        group = cls.construct_instance(group, cleaned_input)
+        cls.clean_instance(info, group)
+        cls.save(info, group, cleaned_input)
+        cls.update_group_permissions_and_users(group, cleaned_input)
         return cls(group=group)
 
     @classmethod
-    def prepare_user_pks(cls, info, group, **data):
-        cleaned_input = cls.clean_input(info, group, data, Group)
-        user_ids: List[str] = cleaned_input["users"]
+    def update_group_permissions_and_users(
+        cls, group: auth_models.Group, cleaned_input: dict
+    ):
+        if "add_users_pks" in cleaned_input:
+            group.user_set.add(*cleaned_input["add_users_pks"])
+        remove_users_pks = cls.get_user_pks(cleaned_input, "remove_users")
+        group.user_set.remove(*remove_users_pks)
+
+        if "add_permissions" in cleaned_input:
+            group.permissions.add(*cleaned_input["add_permissions"])
+        if "remove_perissions" in cleaned_input:
+            remove_permissions = get_permissions(cleaned_input["remove_perissions"])
+            group.permissions.remove(*remove_permissions)
 
-        user_pks = [
-            from_global_id_strict_type(user_id, only_type=User, field="id")
-            for user_id in user_ids
-        ]
+    @classmethod
+    def clean_input(
+        cls, info, instance, data,
+    ):
+        cleaned_input = super().clean_input(info, instance, data)
+        errors = defaultdict(list)
+        for field in ["users", "permissions"]:
+            cls.check_for_duplicates(errors, cleaned_input, field)
```

Comment:
I think it would be more readable if we had it more explicitly:
```
cls.check_for_duplicates(cleaned_input, "add_permissions", "remove_permissions")
cls.check_for_duplicates(cleaned_input, "add_users", "remove_users")
```
It could be moved before the initial `super().clean_input()` call because we can find out pretty quickly that you have duplicates and raise the validation error early.